src/Infrastructure/OpcUa/OpcSession.cs

```diff
@@ -35,14 +34,13 @@ namespace OMP.Connector.Infrastructure.OpcUa
     public class OpcSession : IOpcSession
     {
         private readonly OpcUaConfiguration _opcUaSettings;
-        private readonly IOpcSessionReconnectHandlerFactory _opcSessionReconnectHandlerFactory;
         private readonly ILoggerFactory _loggerFactory;
         private readonly ILogger _logger;
         private readonly ApplicationConfiguration _applicationConfiguration;
         private bool _disposed;
         private SemaphoreSlim _opcSessionSemaphore;
         private CancellationTokenSource _sessionCancellationTokenSource;
-        private IOpcSessionReconnectHandler _reconnectHandler;
+        private SessionReconnectHandler _sessionReconnectHandler;
```

Comment:
Why move away from the interface? This introduces hard coupling again with no way to easily extend it.

Reply:
Restored the interface and wrapper